TidepoolOnboarding/Views/Sections/YourDevicesViews.swift

```diff
@@ -10,38 +10,204 @@ import SwiftUI
+    @State private var isAlertPresented = false
+
+    @State private var isCGMManagerOnboarded = false
+    @State private var cgmManagerViewController: CGMManagerViewController?
+    @State private var isPumpManagerOnboarded = false
+    @State private var pumpManagerViewController: PumpManagerViewController?
+    @State private var isSheetPresented = false
+    @State private var onSheetDismiss: (() -> Void)?
+
+    var body: some View {
+        OnboardingSectionPageView(section: .yourDevices) {
+            PageHeader(title: LocalizedString("Pairing Your Devices", comment: "Onboarding, Your Devices section, Pairing Your Devices view, title"))
+            Paragraph(LocalizedString("Use your product instructions along with this app to help you pair your devices.", comment: "Onboarding, Your Devices section, Pairing Your Devices view, paragraph 1"))
+            Paragraph(LocalizedString("Before pairing your devices, make sure your smart device, on which you are reading this screen, is connected to the internet.", comment: "Onboarding, Your Devices section, Pairing Your Devices view, paragraph 2"))
+            Paragraph(LocalizedString("You must have both CGM and Pump with you to proceed.", comment: "Onboarding, Your Devices section, Pairing Your Devices view, paragraph 3"))
+                .bold()
+            Paragraph(LocalizedString("If you do not yet have both devices, you should wait until you do to proceed.", comment: "Onboarding, Your Devices section, Pairing Your Devices view, paragraph 4"))
+            VStack(alignment: .leading, spacing: 30) {
+                DeviceView(number: 1, attributed: cgmManagerText, checked: isCGMManagerOnboarded)
+                DeviceView(number: 2, attributed: pumpManagerText, checked: isPumpManagerOnboarded)
+            }
+            .padding(.vertical)
+        }
+        .backButtonHidden(true)
+        .nextButtonTitle(nextButtonTitle)
+        .nextButtonAction(nextButtonAction)
+        .alert(isPresented: $isAlertPresented) { alert }
+        .sheet(isPresented: $isSheetPresented, onDismiss: onSheetDismiss) { sheet }
+        .onAppear {
+            isCGMManagerOnboarded = onboardingViewModel.isCGMManagerOnboarded
+            isPumpManagerOnboarded = onboardingViewModel.isPumpManagerOnboarded
+        }
+    }
+
+    private var cgmManagerText: String {
+        return String(format: LocalizedString("Pair your Continuous Glucose Monitor (CGM): <b>%1$@</b>", comment: "Onboarding, Your Devices section, Pairing Your Devices view, list, CGM (1: CGM title)"),
+                      onboardingViewModel.cgmManagerTitle)
+    }
+
+    private var pumpManagerText: String {
+        return String(format: LocalizedString("Pair your Pump: <b>%1$@</b>", comment: "Onboarding, Your Devices section, Pairing Your Devices view, list, pump (1: pump title)"),
+                      onboardingViewModel.pumpManagerTitle)
+    }
+
+    private var nextButtonTitle: String? {
```

Comment:
[nit] The next few functions have tiny bits of logic that might benefit from unit test coverage.  It's not that bad, but it seems like it would benefit from some coverage.  Again, it's pretty simple stuff, and if you think it might be easy to do, I recommend it.  But I realize that moving it all to the `OnboardingViewModel` will just bulk up that (already pretty big) class, and creating a separate ViewModel may seem overkill (especially since we just talked about avoiding unnecessary ViewModels).  So, I'll leave it to you to decide if this logic could be brought under unit test, and if so, how to do it.  Thanks!

Reply:
I can see how to bring `nextButtonTitle` into the view model for testing, but I don't see much benefit (as it is just one simple conditional statement with two different strings).

I really don't see how to bring `nextButtonAction`, `onboardingCGMManager`, or `onboardPumpManager` into the view model without pulling in a significant chunk of the view implementation as well (`alertMessage`, `isAlertPresented`, `cgmManagerViewController`, `pumpManagerViewController`, `isSheetPresented`, and `onSheetDismiss`). Seems like a lot of work for only a limited amount of test coverage.

Reply:
ok